### PR TITLE
Issue-14: replace outline variables in docstrings

### PR DESCRIPTION
**Pull request by ptrimble**

### CODE REVIEW

**Comment by ptrimble** [comment]
I will be going over the 'multiple examples' problem separately (assuming that I am correct that it is an actual problem).

File: gocumber_test.go [hunk 1]
```diff
@@ -201,3 +201,19 @@ func TestExec_NoMatch(t *testing.T) {
...
 }
+
+func TestDocstrings_OutlineVariables(t *testing.T) {
+	steps := make(Definitions)
+	tt := new(testing.T)
+
+	var called bool
+	steps.Given("something is:", func(_ []string, step StepNode) {
+		called = true
+		assert.Equal(t, "minimally functional\n\n", step.PyString().String())
```
Comment (cbandy) [inline gocumber_test.go]: Two newlines is incorrect. If one comes from the initial parsing<sup>1</sup>, I'm fine with leaving it. *Adding* any is definitely not good.

<sup>1</sup> Gherkin tests show the final newline being omitted, but that should be done over in go-gherkin, I think.
 \- https://github.com/cucumber/gherkin3/blob/master/testdata/good/docstrings.feature.ast#L12-L16
 \- https://github.com/cucumber/gherkin3/blob/master/testdata/good/docstrings.feature.ast.json#L38

Comment (ptrimble) [inline gocumber_test.go]: This has to do with how go-gherkin prints out the lines via the `String()` function: https://github.com/muhqu/go-gherkin/blob/e07b6f9917cc9d90dd0a5d5f9ab9a63c82efe045/nodes/nodes.go#L463-L469

In reality what is stored in with the single \n. It's only when we print it in the test that is shows up with two \n's.

I'm at a bit of a loss. It seems like when we execute the step we will have the correct single \n character. I could resolve this like so:

```go
pyString.AddLine(replace(strings.Trim(original.PyString().String(), "\n")))
```

But wouldn't I be removing the newline that will be present in the steps?

Comment (cbandy) [inline gocumber_test.go]: Oh, I didn't realize that `PyStringNode` stored separate lines. We need to keep doing that, so I guess iterate `Lines()` and `AddLine(replace(line))` many times.

Comment (cbandy) [inline gocumber_test.go]: Maybe the assertion should change to `Lines()` as well.

Comment (ptrimble) [inline gocumber_test.go]: Addressed in https://github.com/sittercity/gocumber/commit/835b954deabac4dd01742868bb795de971a0ab78

**Comment by ptrimble** [comment]
Failing build! Whattttttt

**Comment by ptrimble** [comment]
Oh, just a temporary connection issue.

**Comment by cbandy** [comment]
:thumbsup:

**Pull request merged**